File: src/main/java/com/example/jpa/practice/Event.java
package com.example.jpa.practice;

import com.example.jpa.enums.EventType;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

/**
 * @Builder 를 쓰면 Set 과 같은 필드가 null 로 들어간다.
 */
@NamedEntityGraph(
        name = "Event.withEnrollments",
        attributeNodes = @NamedAttributeNode("enrollments")
)
@Entity
@Getter @Setter @EqualsAndHashCode(of = "id")
public class Event {

    @Id @GeneratedValue
    private Long id;

    /**
     * study_id bigint, (FK)
     */
    @ManyToOne // Many 의 테이블에 FK 생성
    private Study study;

    /**
     * 모임 생성한 회원 정보
     */
    @ManyToOne
    private Account createdBy; // timestamp

    @Column(nullable = false)
    private String title;

    @Lob
    private String description; // clob

    @Column(nullable = false)
    private LocalDateTime createdDateTime; // timestamp

    @Column
    private Integer limitOfEnrollments; // integer

    /**
     * event_id (FK)
     */
    @OneToMany(mappedBy = "event")
    @OrderBy("enrolledAt")
    private List<Enrollment> enrollments = new ArrayList<>();

    /**
     * varchar(255) 이지만 아래 enum 의 값만 들어가겠다 STRING 자체로
     */
    @Enumerated(EnumType.STRING)
    private EventType eventType;
}

File: kotlin-project/src/main/kotlin/com/studyolle/domain/Event.kt
package com.studyolle.domain

import com.studyolle.domain.enums.EventType
import java.time.LocalDateTime
import javax.persistence.*

@Entity
class Event(
    @Id
    @GeneratedValue
    val id: Long? = null,

    @ManyToOne
    val study: Study? = null,

    /**
     * 모임 생성한 회원 정보
     */
    /* event 테이블의 컬럼 : created_by_id */
    @ManyToOne
    val createdBy: Account? = null,

    @Column(nullable = false)
    val title: String? = null,

    @Lob
    val description: String? = null,

    @Column(nullable = false)
    val createdDateTime: LocalDateTime? = null,

    @Column(nullable = false)
    val endEnrollmentDateTime: LocalDateTime? = null,

    @Column(nullable = false)
    val startDateTime: LocalDateTime? = null,

    @Column(nullable = false)
    val endDateTime: LocalDateTime? = null,

    @Column
    val limitOfEnrollments: Int? = null,

    /**
     * 참여정보
     */
    @OneToMany(mappedBy = "event")
    @OrderBy("enrolledAt")
    val enrollments: MutableList<Enrollment> = ArrayList(),

    /**
     * EvenType
     * FCFS 선착순
     * CONFIRMATIVE 관리자 확인 필요
     */
    @Enumerated(EnumType.STRING)
    val eventType: EventType? = null,
) {
}